### PR TITLE
Added Centrifuge rule

### PR DESCRIPTION
**Pull request by qiyunzhu**
A few notes:

1) Although we are still unsure if it is recommended to reestimate per-rank relative abundance using Bracken on Centrifuge output, I chose to do so for now, after talking with Gabe and based on my understanding.

2) The organization of config.yaml and parser may be modified to be more generic rather than specific. I didn't make any change in this PR, but we can discuss on it.

### CODE REVIEW

**Comment by tanaes** [comment]
Awesome, thank! Will take a look.

**Comment by qiyunzhu** [comment]
@sjanssen2 @mortonjt Wanna take a look? Thanks!

File: oecophylla/taxonomy/taxonomy.rule [hunk 1]
```diff
@@ -200,19 +201,124 @@ rule taxonomy_kraken_combine_profiles:
...
-        if params['levels']:
-            for level in params['levels'].split(','):
-                redists = ['%s.redist.%s.txt' % (x[:-12], level)
-                           for x in input]
-                pandas2biom('%s_redist.%s.biom' % (output[0][:-13], level),
-                            combine_bracken(zip(samples, redists)))
+        for level in params['levels'].split(','):
+            redists = ['%s.redist.%s.txt' % (x[:-12], level)
+                       for x in input]
+            pandas2biom('%s_redist.%s.biom' % (output[0][:-13], level),
```
Comment (mortonjt) [inline oecophylla/taxonomy/taxonomy.rule]: I'm not too crazy about these sorts of magic numbers.  What is column `13`?

File: oecophylla/taxonomy/taxonomy.rule [hunk 2]
```diff
@@ -200,19 +201,124 @@ rule taxonomy_kraken_combine_profiles:
...
     run:
         pandas2biom(output[0], combine_profiles(zip(samples, input)))
-        if params['levels']:
-            for level in params['levels'].split(','):
-                redists = ['%s.redist.%s.txt' % (x[:-12], level)
-                           for x in input]
-                pandas2biom('%s_redist.%s.biom' % (output[0][:-13], level),
-                            combine_bracken(zip(samples, redists)))
+        for level in params['levels'].split(','):
+            redists = ['%s.redist.%s.txt' % (x[:-12], level)
```
Comment (mortonjt) [inline oecophylla/taxonomy/taxonomy.rule]: Same thing here

**Review by mortonjt — APPROVED** [review]
Looks good!  Just to confirm - the wet run works right?

File: oecophylla/taxonomy/taxonomy.rule (hunk 1 above)
Comment (tanaes) [reply oecophylla/taxonomy/taxonomy.rule]: agreed. Possible to parse table headers and address with labels?

File: oecophylla/taxonomy/taxonomy.rule [hunk 3]
```diff
@@ -200,19 +201,124 @@ rule taxonomy_kraken_combine_profiles:
...
+                       for x in input]
+            pandas2biom('%s_redist.%s.biom' % (output[0][:-13], level),
+                        combine_bracken(zip(samples, redists)))
+
+
+rule centrifuge:
+    input:
+        taxonomy_dir + "centrifuge/combined_profile.biom"
```
Comment (tanaes) [inline oecophylla/taxonomy/taxonomy.rule:324]: can also change to `rules.taxonomy_centrifuge_combine_profiles.output` 

This way if you update that output don't need to catch the change in import!

File: oecophylla/taxonomy/taxonomy.rule [hunk 4]
```diff
@@ -328,4 +433,5 @@ rule taxonomy:
     input:
         taxonomy_dir + "metaphlan2/combined_profile.biom",
         taxonomy_dir + "kraken/combined_profile.biom",
+        taxonomy_dir + "centrifuge/combined_profile.biom",
         taxonomy_dir + "shogun/combined_profile.biom"
```
Comment (tanaes) [inline oecophylla/taxonomy/taxonomy.rule:444]: similar to above comment -- calling from `rules.[rule_name].output` simplifies maintenance and updating.

**Review by tanaes — APPROVED** [review]
Looks great! Can update top-level rule input style if desired. What do you think?

**Comment by qiyunzhu** [comment]
@mortonjt I guess "wet run" means running it locally with actual profiling operation enabled. The answer is yes. All my PRs were tested and worked locally.

File: oecophylla/taxonomy/taxonomy.rule (hunk 1 above)
Comment (qiyunzhu) [reply oecophylla/taxonomy/taxonomy.rule]: This `[:-13]` is to remove `_profile.biom` from `xxx_profile.biom`. Is there a better way for doing so?

File: oecophylla/taxonomy/taxonomy.rule (hunk 2 above)
Comment (qiyunzhu) [reply oecophylla/taxonomy/taxonomy.rule]: See my response below.

**Comment by tanaes** [comment]
Have you seen
https://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.str.replace.html?
I like to use it for that sort of operation.

File: oecophylla/taxonomy/taxonomy.rule (hunk 3 above)
Comment (qiyunzhu) [reply oecophylla/taxonomy/taxonomy.rule:324]: Jon I didn't quite get your rationale. Can you elaborate little bit?

**Comment by qiyunzhu** [comment]
@tanaes Thanks! I know this Pandas trick. But I was working on a simple Python string, not a Pandas.

**Comment by qiyunzhu** [comment]
Hi @mortonjt look at new lines #311-316. Do you like this style?
(sorry I just saw your reply)

**Review by mortonjt — APPROVED** [review]

**Comment by qiyunzhu** [comment]
@tanaes @mortonjt Feel free to click "merge" :)

**Pull request merged**